addon/components/column-visibility-panel.js

```diff
@@ -0,0 +1,29 @@
+import Component from '@ember/component';
+
+export default Component.extend({
+  displayedColumnsPanel: false,
+  hasToggleableColumns: false,
+
+  clickHandler(event) {
+    if (event.target.closest('.button-column-visibility-panel')) {
+      this.toggleProperty('displayedColumnsPanel');
+    }
+    else if (!event.target.closest('.side-panel--appearance')) {
+      this.set('displayedColumnsPanel', false);
+    }
+  },
+
+  init(){
+    this._super();
+
+    this.set('bindClickHandler', this.clickHandler.bind(this));
```

Comment:
my bad i meant `this.set('clickHandler', this.handleClick.bind(this))`

Reply:
emberjs component have event based method maybe we can simply use it ?
```js
click(event) {
    if (event.target.closest('.button-column-visibility-panel')) {
      this.toggleProperty('displayedColumnsPanel');
    } else if (!event.target.closest('.side-panel--appearance')) {
      this.set('displayedColumnsPanel', false);
    }
}
```

Reply:
My bad, the click is on the document :) forget my previous message